### PR TITLE
txn: pessimistic lock returns multiple values

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

###  What have you changed?
Return values when acquiring pessimistic locks successfully. In some cases, it can reduce get RPCs and it brings about 10% improvement in TPC-C benchmark.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

###  Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/issues/6680

### CODE REVIEW

**Comment by youjiali1995** [comment]
@coocood

**Comment by coocood** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -1460,7 +1460,16 @@ txn_command_future!(future_prewrite, PrewriteRequest, PrewriteResponse, (v, resp
     resp.set_errors(extract_key_errors(v).into())
 });
 txn_command_future!(future_acquire_pessimistic_lock, PessimisticLockRequest, PessimisticLockResponse, (v, resp) {
-    resp.set_errors(extract_key_errors(v).into())
+    match v {
+        Ok(Ok(res)) => match res {
+            PessimisticLockRes::Values(values) => {
```
Comment (nrc) [inline src/server/service/kv.rs]: Could this match be factored out into a method on `PessimisticLockResult`?

File: src/storage/txn/process.rs [hunk 2]
```diff
@@ -606,22 +606,32 @@ fn process_write_impl<S: Snapshot, L: LockManager>(
             is_first_lock,
             for_update_ts,
             wait_timeout,
+            return_values,
             ..
         }) => {
             let mut txn = MvccTxn::new(snapshot, start_ts, !cmd.ctx.get_not_fill_cache());
-            let mut locks = vec![];
             let rows = keys.len();
+            let mut res = if !return_values {
```
Comment (nrc) [inline src/storage/txn/process.rs]: style nit: switch the branches and remove the `!`

**Comment by youjiali1995** [comment]
> LGTM but a concern that using this interface we cannot distinguish a non-existing key from a pair with empty value. This does not affect the use case of TiDB but may cause some trouble if we will extend pessimistic transactions to pure kv services.
> (We don't need to solve it at the moment IMO)

Yes. I can file an issue now.

**Pull request merged**